dandiapi/api/views/zarr.py

```diff
@@ -169,3 +174,53 @@ def delete_files(self, request, zarr_id):
             paths = [file['path'] for file in serializer.validated_data]
             zarr_archive.delete_files(paths)
         return Response(None, status=status.HTTP_204_NO_CONTENT)
+
+
+@swagger_auto_schema(
+    method='GET',
+    responses={
+        200: 'foobar',
+    },
+    manual_parameters=[
+        openapi.Parameter(
+            'path',
+            openapi.IN_PATH,
+            'a file or directory path within the zarr file.',
+            type=openapi.TYPE_STRING,
+            required=True,
+            pattern=r'.*',
+        )
+    ],
+)
+@api_view(['GET'])
+def explore_zarr_archive(request, zarr_id: str, path: str):
+    zarr_archive = get_object_or_404(ZarrArchive, zarr_id=zarr_id)
+    if path == '':
+        path = '/'
+    # If a path ends with a /, assume it is a directory.
+    # Return a JSON blob which contains URLs to the contents of the directory.
+    if path.endswith('/'):
+        # Strip off the trailing /, it confuses the ZarrChecksumFileUpdater
+        path = path[:-1]
```

Comment:
I generally prefer `path = path.rstrip('/')` since it's more explicit, but this is beyond nitpicking :smile: and I don't have a strong opinion.